### PR TITLE
fix: speed up help by 2x on `sam` binary

### PR DESCRIPTION
**Pull request by sriram-mv**
- Only import cli commands for specific invocations of non-root
  commands, eg: `sam build` not for `sam`

### BEFORE

Screen Shot 2022-09-07 at 7 39 33 PM

### AFTER

Screen Shot 2022-09-07 at 7 40 23 PM

#### Which issue(s) does this change fix?
Speed up help for aws sam cli.

#### Why is this change necessary?
The `--help` page which is the first command that arguably a user would load for their first experience with aws sam cli. This command should be blazingly fast.

#### How does it address the issue?
Only import modules if the child commands such as `sam build --help` etc are loaded, otherwise load a generic command with a name and a short help. 

#### What side effects does this change have?
N/A, This will probably break a bunch of tests that I will need to fix.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by mndeveci** [comment]
LGTM in general. I think we should have a sanity test to validate we don't have the error below;
```
"Command %s is not configured correctly. It must expose an function called 'cli'"
```

So that when someone adding a new command they can catch their error during `make pr`.

**Comment by sriram-mv** [comment]
> LGTM in general. I think we should have a sanity test to validate we don't have the error below;
> 
> ```
> "Command %s is not configured correctly. It must expose an function called 'cli'"
> ```
> 
> So that when someone adding a new command they can catch their error during `make pr`.

test_get_command_must_skip_on_absence_of_cli_method does what you are asking, is there something else you are poking at perhaps?

File: samcli/cli/command.py [hunk 1]
```diff
@@ -29,6 +29,21 @@
...
+_SAM_CLI_COMMAND_SHORT_HELP = {
+    "init": "Init an AWS SAM application",
+    "validate": "Validate an AWS SAM template",
+    "build": "Build your Lambda Function code",
+    "local": "Run your serverless function locally",
+    "package": "Package an AWS SAM application",
+    "deploy": "Deploy an AWS SAM application",
+    "delete": "Delete an AWS SAM application and the artifacts created by sam deploy",
+    "logs": "Fetch logs for a function",
+    "publish": "Publish a packaged AWS SAM template to AWS Serverless Application Repository",
```
[the remark this answers is not in the document]
Comment (sriram-mv) [reply samcli/cli/command.py:41]: Yes, if you run `sam publish --help`

**Review by hoffa — APPROVED** [review]

File: samcli/cli/command.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hoffa) [reply samcli/cli/command.py:41]: I see, they could probably refer to the same string so we don't have to type them twice.

[the remark this answers is not in the document]
Comment (sriram-mv) [reply samcli/cli/command.py:41]: yeah, we could move them to a common spot, when I typed this PR out, I was raging against imports in general haha.

**Review by mndeveci — APPROVED** [review]

**Pull request merged**